src/power_grid_model/core/buffer_handling.py

```diff
@@ -231,9 +230,7 @@ def get_buffer_view(data: np.ndarray | Mapping[str, np.ndarray], schema: Compone
     return _get_sparse_buffer_view(data, schema)
 
 
-def create_buffer(
-    properties: BufferProperties, schema: ComponentMetaData
-) -> np.ndarray | dict[ComponentType, np.ndarray]:
+def create_buffer(properties: BufferProperties, schema: ComponentMetaData) -> np.ndarray | dict[str, np.ndarray]:
```

Comment:
this was an actual bug: `buffer` has type `np.ndarray` or sparse buffer, which has type `{"data": np.ndarray, "indptr": np.ndarray}`. It should not have been `ComponentType`

Reply:
Should we introduce another enum :P